eva-pipeline/src/main/java/embl/ebi/variation/eva/pipeline/annotation/load/VariantAnnotationLineMapper.java

```diff
@@ -0,0 +1,376 @@
+/*
+ * Copyright 2015-2016 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package embl.ebi.variation.eva.pipeline.annotation.load;
+
+import org.apache.commons.lang.ArrayUtils;
+import org.opencb.biodata.models.variant.annotation.ConsequenceType;
+import org.opencb.biodata.models.variant.annotation.Score;
+import org.opencb.biodata.models.variant.annotation.VariantAnnotation;
+import org.opencb.biodata.models.variation.PopulationFrequency;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.batch.item.file.LineMapper;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * @author Diego Poggioli
+ *
+ * Map a line in VEP output file to {@link VariantAnnotation}
+ *
+ * Example of VEP output line
+ * 20_60343_G/A	20:60343	A	-	-	-	intergenic_variant	-	-	-	-	-	-
+ * 20_60419_A/G	20:60419	G	-	-	-	intergenic_variant	-	-	-	-	-	-
+ * 20_60479_C/T	20:60479	T	-	-	-	intergenic_variant	-	-	-	-	-	rs149529999	GMAF=T:0.0018;AFR_MAF=T:0.01;AMR_MAF=T:0.0028
+ * 20_60523_-/C	20:60522-60523	C	-	-	-	intergenic_variant	-	-	-	-	-	rs150241001	GMAF=C:0.0115;AFR_MAF=C:0.05;AMR_MAF=C:0.0028
+ *
+ * Please note that most of the code is from org.opencb.biodata.formats.annotation.io.VepFormatReader
+ * public methods in VepFormatReader can't be reused because there is a reference to the previous line (currentVariantString)
+ * that prevent each line to be independent
+ *
+ * Here each line is mapped to {@link VariantAnnotation}; in case of two annotations for the same variant, a new
+ * {@link VariantAnnotation} object is created containing only the fields that will be appended:
+ *  - ConsequenceTypes
+ *  - Hgvs
+ */
+public class VariantAnnotationLineMapper implements LineMapper<VariantAnnotation> {
+    private static final Logger logger = LoggerFactory.getLogger(VariantAnnotationLineMapper.class);
+
+    /**
+     * Map a line in VEP output file to {@link VariantAnnotation}
+     * @param line in VEP output
+     * @param lineNumber
+     * @return a {@link VariantAnnotation}
+     * @throws Exception
+     *
+     * Most of the code is from org.opencb.biodata.formats.annotation.io.VepFormatReader#read() with few differences:
+     *  - An empty array is initialized for Hgvs (like ConsequenceTypes);
+     *  - parseFrequencies is always true and the all line is always parsed;
```

Comment:
Could the fact that the whole line is parsed allow to load more fields in the future?